File: client-py/src/utils/RowRecord.py
from IoTDBConstants import TSDataType
from Field import Field


class RowRecord(object):

    def __init__(self, timestamp, field_list=None):
        self.__timestamp = timestamp
        self.__field_list = field_list

    def add_field(self, field):
        self.__field_list.append(field)

    def add_field(self, value, data_type):
        self.__field_list.append(Field.get_field(value, data_type))

    def __str__(self):
        str_list = [str(self.__timestamp)]
        for field in self.__field_list:
            str_list.append("\t\t")
            str_list.append(str(field))
        return "".join(str_list)

    def get_timestamp(self):
        return self.__timestamp

    def set_timestamp(self, timestamp):
        self.__timestamp = timestamp

    def get_fields(self):
        return self.__field_list

    def set_fields(self, field_list):
        self.__field_list = field_list

    def set_field(self, index, field):
        self.__field_list[index] = field

File: client-go/utils/RowRecord.go
package utils

import "fmt"

type RowRecord struct {
	timestamp int64
	fieldList []Field
}

func NewRowRecord(timestamp int64, fieldList []Field) *RowRecord {
	r_ := &RowRecord{}
	r_.timestamp = timestamp
	r_.fieldList = fieldList
	return r_
}

func (r_ *RowRecord) AddField(f Field) {
	r_.fieldList = append(r_.fieldList, f)
}

func (r_ *RowRecord) AddFieldWithValue(dataType int32, value interface{}) {
	r_.fieldList = append(r_.fieldList, *NewField(dataType, value))
}

func (r_ *RowRecord) ToString() string {
	str := string(r_.timestamp)
	for _, v := range r_.fieldList {
		str += "\t\t" + v.ToString()
	}
	return str
}

func (r_ *RowRecord) SetTimestamp(timestamp int64) {
	r_.timestamp = timestamp
}

func (r_ *RowRecord) GetTimestamp() int64 {
	return r_.timestamp
}

func (r_ *RowRecord) SetField(index int, field Field) bool {
	if index < len(r_.fieldList) {
		r_.fieldList[index] = field
		return true
	}
	fmt.Println("Out of FieldList Length!")
	return false
}

func (r_ *RowRecord) SetFields(fields []Field) {
	r_.fieldList = fields
}

func (r_ *RowRecord) GetFields() []Field {
	return r_.fieldList
}
